rpmio/rpmkeyring.cc

```diff
@@ -133,7 +148,7 @@ int rpmKeyringModify(rpmKeyring keyring, rpmPubkey key, rpmKeyringModifyMode mod
 	if (item->second->fp == key->fp)
 	    break;
     }
-    if (item != range.second && (mode == RPMKEYRING_DELETE || mode == RPMKEYRING_REPLACE)) {
+    if (item != range.second) {
```

Comment:
Hmm, shouldn't this still check that the mode is actually DELETE?

Reply:
Hmm, guess not. The code just looks a bit weird.

Reply:
Yeah, this deletes the old entry - for both ADD and DELETE